### ISSUE TITLE
suggestion: check model changes in after_commit

### ISSUE THREAD

**Issue by shwetabhgarg** [ankane/searchkick#948]
**TLDR**: check if model has any changes before calling reindex on after_commit

**Long version for unambiguity**:
Searchkick reindexes the model on after_commit, but this need not be done if there are no changes to the model. For example, imagine a query like
`City.where(name: "New York").update(name: "New York")`
Such a query can be generated by a code that is dynamically building the update parameters. Rails  is clever enough to realise that no update database query is required. However, it still results in an empty transaction, something like this:
```
(0.2ms)  BEGIN
(0.2ms)  COMMIT
```
This triggers the after_commit to Searchkick. It results in queries for fetching loaded model associations, and then reindexing elastic search with all the data. Since there is no data change here, this affects performance of the system for no benefit at all.

A possible solution would be to check if `model.previous_changes.any?` before running reindex on after_commit.
(If one wants to be more thorough still, one could also compare first and last in model.previous_changes in case there are multiple updates to a model object in a transaction. But even without this, the `.any?` check should still eliminate most cases.)

**Comment by ankane** [comment]
Hey @shwetabhgarg, thanks for the suggestion. In this case, I think it's best for the app to not trigger `BEGIN`/`COMMIT` in this first place.
